java/util/HashMap/HashMapTest.cpp

```diff
@@ -51,95 +53,111 @@ TEST (JavaUtil, HashMapConstructor) {
 	HashMap<String, String> hashMap = container;
 
 	// Test valid size()
-	expectedSize = 2;
-	ASSERT_EQUAL(expectedSize, hashMap.size());
+	ASSERT_EQUAL(2, hashMap.size());
 	ASSERT_FALSE(hashMap.isEmpty());
 
 	// Test valid data between hashMap and container
-	String expectedValue = container.get("sample");
-	String actualValue = hashMap.get("sample");
+	expected = container.get("sample");
+	actual = hashMap.get("sample");
 
-//    ASSERT_FALSE(expectedValue.isNull());
-//    ASSERT_FALSE(actualValue.isNull());
-	ASSERT_STR(expectedValue.toString(), actualValue.toString());
+	ASSERT_FALSE(expected.isEmpty());
+	ASSERT_FALSE(actual.isEmpty());
+	ASSERT_STR(expected.toString(), actual.toString());
 
-	expectedValue = container.get("key");
-	actualValue = hashMap.get("key");
+	expected = container.get("key");
+	actual = hashMap.get("key");
 
-//    ASSERT_FALSE(expectedValue.isNull());
-//    ASSERT_FALSE(actualValue.isNull());
-	ASSERT_STR(expectedValue.toString(), actualValue.toString());
+	ASSERT_FALSE(expected.isEmpty());
+	ASSERT_FALSE(actual.isEmpty());
+	ASSERT_STR(expected.toString(), actual.toString());
 }
 
-TEST (JavaUtil, HashMapClear) {
+TEST(JavaUtil, HashMapClear) {
 	// Given valid hashMap to test clear()
 	HashMap<Long, Integer> hashMap;
-	hashMap.put(100, 25);
-	hashMap.put(500, 123);
+	hashMap.put((long) 100, 25);
+	hashMap.put((long) 500, 123);
+	ASSERT_EQUAL(2, hashMap.size());
 
 	hashMap.clear();
-	int expectedSizeAfterClear = 0;
-	ASSERT_EQUAL(expectedSizeAfterClear, hashMap.size());
+	ASSERT_EQUAL(0, hashMap.size());
 }
 
-TEST (JavaUtil, HashMapClone) {
-	// Given valid hash map to test clone()
+TEST(JavaUtil, HashMapClone) {
+	String actualKey;
+	String expected;
+	String actual;
+
+	// Create a HashMap to test to test clone()
 	HashMap<String, String> hashMap;
-	hashMap.put("some key", "123");
-	hashMap.put("another thing and key", "-44444");
-	HashMap<String, String> anotherMap = hashMap.clone();
-
-	// Given valid key/value to test and make sure that result is not null
-	String expectedKey = "some key";
-	String expectedValue = "123";
-	String result = anotherMap.get(expectedKey);
-	ASSERT_STR(expectedValue.toString(), result.toString());
-
-	// Given valid key/value to test and make sure that result is not null
-	expectedKey = "another thing and key";
-	expectedValue = "-44444";
-	result = anotherMap.get(expectedKey);
-	ASSERT_STR(expectedValue.toString(), result.toString());
-
-	// Given invalid key and make result is null
-	String notExpectedKey = "wrong key";
-	result = anotherMap.get(notExpectedKey);
+	hashMap.put("Key1", "Value of Key1");
+	hashMap.put("Key2", "Value of Key2");
+	HashMap<String, String> anotherMap = (HashMap<String, String>) hashMap.clone();
```

Comment:
Please remove this cast, is't not necessary

Reply:
Thanks. Fixed.